virtualizarr/tests/test_integration.py

```diff
@@ -126,6 +127,15 @@ def roundtrip_as_in_memory_icechunk(vds: xr.Dataset, tmpdir, **kwargs):
     return xr.open_zarr(session.store, zarr_format=3, consolidated=False, **kwargs)
 
 
+def roundtrip_as_virtual_object_store(vds: xr.Dataset, tmpdir, **kwargs):
+    from obstore.store import LocalStore
+
+    store = LocalStore()
+    stores = {"file://": store}
+    # Convert dataset to a be obstore backed for loading
+    return vds.virtualize.to_xarray(stores)
```

Comment:
This new `storage` module deserves it's own test module, where you directly test the `ManifestStore` interface behaves as expected. Otherwise we're going to end up with every test in `test_integration.py`. 

I also heard there is a testing scaffold in zarr-python that we could use to help with testing the `Store` interface is correct too.